core/src/components/item-sliding/item-sliding.tsx

```diff
@@ -407,16 +407,17 @@ export class ItemSliding implements ComponentInterface {
   }
 
   private setOpenAmount(openAmount: number, isFinal: boolean) {
+    const { el } = this;
+
     if (this.tmr !== undefined) {
       clearTimeout(this.tmr);
+      el.classList.remove('item-sliding-closing');
```

Comment:
There was another edge case I did not catch during our debug session. It looks like we clear the timeout in 2 different places: https://github.com/ionic-team/ionic-framework/blob/9448783bb19b187f867054c86d215e3dc97952c1/core/src/components/item-sliding/item-sliding.tsx#L315-L316

As a result, this bug can appear in two different ways. To fix this properly, we can make a `clearCloseTimeout` method that runs the following code:

```js
if (this.tmr !== undefined) {
  clearTimeout(this.tmr);
  this.tmr = undefined;
  this.el.classList.remove('item-sliding-closing');
}
```

From here, we'd need to call `this.clearCloseTimeout()` in [setOpenAmount](https://github.com/ionic-team/ionic-framework/blob/9448783bb19b187f867054c86d215e3dc97952c1/core/src/components/item-sliding/item-sliding.tsx#L410) and [onStart](https://github.com/ionic-team/ionic-framework/blob/9448783bb19b187f867054c86d215e3dc97952c1/core/src/components/item-sliding/item-sliding.tsx#L314).

Reply:
A method was created called `resetCloseState()` to hold the code that is called in two different times and updated the code to call this new method.